consultation_analyser/consultations/public_schema_files/thememapping_schema.json

```diff
@@ -0,0 +1,296 @@
+      },
+      "required": [
+        "question_part",
+        "respondent",
+        "text",
+        "chosen_options"
+      ],
+      "title": "Answer",
+      "type": "object"
+    },
+    "Consultation": {
+      "properties": {
+        "title": {
+          "maxLength": 256,
+          "title": "Title",
+          "type": "string"
+        }
+      },
+      "required": [
+        "title"
+      ],
+      "title": "Consultation",
+      "type": "object"
+    },
+    "ExecutionRun": {
+      "properties": {
+        "type": {
+          "$ref": "#/$defs/Type1"
+        }
+      },
+      "required": [
+        "type"
+      ],
+      "title": "ExecutionRun",
+      "type": "object"
+    },
+    "Framework": {
+      "properties": {
+        "execution_run": {
+          "anyOf": [
+            {
+              "$ref": "#/$defs/ExecutionRun"
+            },
+            {
+              "type": "null"
+            }
+          ],
+          "default": null
+        },
+        "question_part": {
+          "$ref": "#/$defs/QuestionPart"
+        },
+        "change_reason": {
+          "maxLength": 256,
+          "title": "Change Reason",
+          "type": "string"
+        },
+        "user": {
+          "anyOf": [
+            {
+              "$ref": "#/$defs/User"
+            },
+            {
+              "type": "null"
+            }
+          ],
+          "default": null
+        },
+        "precursor": {
+          "anyOf": [
+            {
+              "$ref": "#/$defs/Framework"
+            },
+            {
+              "type": "null"
+            }
+          ],
+          "default": null
+        }
+      },
+      "required": [
+        "question_part",
+        "change_reason"
+      ],
+      "title": "Framework",
+      "type": "object"
+    },
+    "Question": {
+      "properties": {
+        "text": {
+          "title": "Text",
+          "type": "string"
+        },
+        "consultation": {
+          "$ref": "#/$defs/Consultation"
+        },
+        "number": {
+          "title": "Number",
+          "type": "integer"
+        }
+      },
+      "required": [
+        "text",
+        "consultation",
+        "number"
+      ],
+      "title": "Question",
+      "type": "object"
+    },
+    "QuestionPart": {
+      "properties": {
+        "question": {
+          "$ref": "#/$defs/Question"
+        },
+        "text": {
+          "title": "Text",
+          "type": "string"
+        },
+        "type": {
+          "$ref": "#/$defs/Type"
+        },
+        "options": {
+          "anyOf": [
+            {
+              "type": "object"
+            },
+            {
+              "type": "null"
+            }
+          ],
+          "default": null,
+          "description": "JSON data",
+          "title": "Options"
+        },
+        "number": {
+          "title": "Number",
+          "type": "integer"
+        }
+      },
+      "required": [
+        "question",
+        "text",
```

Comment:
Should the `text` be optional for the case where we have a question with one question part? In this case the import takes uses the text field in the question.

Reply:
Hm this is reading from models.py where: 
`text = models.TextField()`, so no null or blank option. Do we allow empty strings there?

We could write an exception particularly for this case in `generate_openapi_yaml.py`, but should we be changing the model?

Reply:
You're right, they can just use the empty string. I think that's actually preferable in terms of model so you don't have to check for `None` or `''`.